innersource-license.md

```diff
@@ -0,0 +1,94 @@
+
+## Context
+
+- A large organization with many legal entities (subsidiaries) that want to share code. When the organization gets larger, the value of this pattern increases.
+- As per definition, the legal entities have their own legal rights and obligations.
+- Multiple of these legal entities are developing software, and are using services of the other legal entities. They have a motivation to contribute to each other’s source code.
+- A sufficient complexity of the organization and its organizational structure
+
+## Forces
+
+- **Level of effort** required to write formal agreements, especially if they need to take into account technical, legal, and business perspectives.
+- A large organization (consisting of many legal entities) has many **internal regulations**. Any new agreements that are made have to comply with these regulations, e.g. security, privacy, procurement processes, etc. The volume of regulations can make it difficult to assess whether sharing software between two legal entities is compliant with these regulations, especially when there is no standard procedure.
+- If any of the legal entities in the organization has a **business model** that depends on proprietary code and accounting of licensing fees within the organization
+- **Company culture** that isn’t used to InnerSource collaboration and sharing code. This results in uncertainty about the rights and obligations when using shared code.
+- Freedom over using the software leads to competition, and spread of ownership
+- There are legal contracts in place which cover the sharing of source code. These contracts are not standardized, so they create additional effort in negotiating and understanding for every project. The existing contracts may also not allow sharing source code in an open enough sense to support a true InnerSource approach.
+- Alternatively, there are no legal contracts in place but source code is shared informally. That might create uncertainty in cases where clarity about ownership and rights and obligations is needed.
+
+## Solutions
+
+Creating an **InnerSource License** customized to the needs of the organization in question (and their legal entities). This license needs to be generic enough to be applied to the most important inter-company relationships.
+
+It is important to write the InnerSource License such that it truly allows for OpenSource-like collaborations across the boundaries of the involved legal entities. Therefore the 4 freedoms of free software should be integrated into the license.
+
+The License is written as a formal legal document, and can be used as part of contracts between the legal entities to govern the code sharing agreements.
+
```

Comment:
We should mention that an InnerSource license needs to be ratified by the responsible bodies in the respective organisation. If there is no such ratification, the license might not be widely accepted by management ... or developers with a high need for job security.

Reply:
It might also make sense to mention to not be too strict about what can be done with the SW. We (at Bosch) made that mistake and precluded the use of InnerSource software in Open Source projects later on. This turned out to be a killer criterium for some of our projects and eventually triggered the development of a new version of the license, which took a lot of effort.